core/src/tangram.h

```diff
@@ -2,144 +2,213 @@
 
 #include "data/properties.h"
 #include "util/ease.h"
+#include <array>
 #include <memory>
-#include <vector>
+#include <mutex>
+#include <queue>
 #include <string>
-
-/* Tangram API
- *
- * Primary interface for controlling and managing the lifecycle of a Tangram
- * map surface
- */
+#include <vector>
 
 namespace Tangram {
 
+class AsyncWorker;
 class DataSource;
+class InputHandler;
+class Labels;
+class Scene;
+class TileManager;
+class TileWorker;
+class View;
+
+struct TouchItem {
+    std::shared_ptr<Properties> properties;
+    float position[2];
+    float distance;
+};
+
+struct SceneUpdate {
```

Comment:
Well... not in this implementation it can't. The `vector<SceneUpdate>` declared below needs to know the size of a `SceneUpdate` (just one reason why it'll be nice to make the `Map` implementation private).